piptools/writer.py

```diff
@@ -189,17 +189,18 @@ def _iter_lines(
                 yield MESSAGE_UNSAFE_PACKAGES
 
             for ireq in unsafe_requirements:
-                req = self._format_requirement(
-                    ireq,
-                    reverse_dependencies,
-                    primary_packages,
-                    marker=markers.get(key_from_ireq(ireq)),
-                    hashes=hashes,
-                )
+                ireq_key = key_from_ireq(ireq)
                 if not self.allow_unsafe:
-                    yield comment("# {}".format(req))
+                    yield comment("# {}".format(ireq_key))
                 else:
-                    yield req
+                    line = self._format_requirement(
```

Comment:
Maybe just `yield` it right away? I mean, it's expected that only lines are yielded by this generator function anyway, so this variable name doesn't add much semantical meaning to the code.

Reply:
Yeah, we can do that. I did it the same way as it's done above (see 171th line). IMO the `yield line` looks more readable than `yield function call with many arguments` and it's debugger-friendly.

Reply:
ok